fw/http.c

```diff
@@ -714,9 +714,11 @@ tfw_http_prep_redir(TfwHttpResp *resp, unsigned short status, TfwStr *rmark,
 	char *date_val = *this_cpu_ptr(&g_buf);
 	char *cl_val = *this_cpu_ptr(&g_buf) + SLEN(S_V_DATE);
 	char *body_val = NULL;
-	TfwStr *proto = &protos[TFW_CONN_PROTO(req->conn) == TFW_FSM_HTTPS];
+	char *host_val = NULL;
+	char *uri_path_val = NULL;
+	TfwStr *proto = &protos[TFW_CONN_PROTO(req->conn) & TFW_FSM_HTTPS];
```

Comment:
This requires that `TFW_FSM_HTTPS` is 1 and if we change it, then we get a memory access violation here.

```suggestion
	TfwStr *proto = &protos[!!(TFW_CONN_PROTO(req->conn) & TFW_FSM_HTTPS)];
```

Also `protos` should be `const`